### PR TITLE
Add --host and --port options

### LINKED ISSUE #27: Would be nice to specify host and port with proper options
I would like to be able to specify alternate host or port on the command line with options like `--host` and `--port`. The `env MPD_HOST=...` syntax gets a little tiresome.

### PR DESCRIPTION
**Pull request by adonSh**
resolves #27

### CODE REVIEW

File: src/args.c [hunk 1]
```diff
@@ -169,6 +179,21 @@ int ashuffle_options(struct ashuffle_options * opts,
                 return -1;
             }
             state = NO_STATE;
+        } else if (state == HOST) {
+            opts->host = malloc(strlen(argv[i]) + 1);
+            if (opts->host == NULL) {
+                fputs("Error processing host\n", stderr);
+                return -1;
+            }
+            snprintf(opts->host, strlen(argv[i]) + 1, "%s", argv[i]);
```
Comment (joshkunz) [inline src/args.c]: snprintf is a complicated tool for a string copy. Just use a `memcpy` here:

```
memcpy(opts->host, argv[i], strlen(argv[i]) + 1);
```

Even better, you could just use `strdup` here, so you don't have to muck about with string lengths.

File: src/args.c [hunk 2]
```diff
@@ -169,6 +179,21 @@ int ashuffle_options(struct ashuffle_options * opts,
...
+            state = NO_STATE;
+        } else if (state == PORT) {
+            if (strlen(argv[i]) >= PORTLEN) {
+                fputs("Error processing port\n", stderr);
```
[the remark this answers is not in the document]
Comment (adonSh) [reply src/args.c]: I think fputs makes more sense here. There was no function call, so no error would be set. I changed the message to be more informative in the next commit.

File: src/args.c (hunk 1 above)
Comment (adonSh) [reply src/args.c]: fixed in 9fa876d

File: src/ashuffle.c [hunk 3]
```diff
@@ -465,6 +466,9 @@ int main (int argc, char * argv[]) {
     }
     list_free(&options.ruleset);
 
+    /* free host string */
```
[the remark this answers is not in the document]
Comment (adonSh) [reply src/ashuffle.c]: Fixed in 9fa876d

File: src/args.h [hunk 4]
```diff
@@ -13,6 +15,8 @@ struct ashuffle_options {
     FILE * file_in;
     bool check_uris;
     unsigned queue_buffer;
+    char * host;
+    char port[PORTLEN];
```
[the remark this answers is not in the document]
Comment (adonSh) [reply src/args.h]: Fixed in 9fa876d

File: src/ashuffle.c [hunk 5]
```diff
@@ -380,16 +380,17 @@ int main (int argc, char * argv[]) {
     /* attempt to connect to MPD */
     struct mpd_connection *mpd;
 
-    /* Attempt to use MPD_HOST variable if available.
-     * Otherwise use 'localhost'. */
-    char * mpd_host_raw = getenv("MPD_HOST") ?
+    /* Attempt to get host from command line if available. Otherwise use
+     * MPD_HOST variable if available. Otherwise use 'localhost'. */
+    char * mpd_host_raw = options.host ? options.host : getenv("MPD_HOST") ?
```
[the remark this answers is not in the document]
Comment (adonSh) [reply src/ashuffle.c]: Fixed in 9fa876d

File: src/args.c [hunk 6]
```diff
@@ -169,6 +179,21 @@ int ashuffle_options(struct ashuffle_options * opts,
                 return -1;
             }
             state = NO_STATE;
+        } else if (state == HOST) {
+            opts->host = malloc(strlen(argv[i]) + 1);
+            if (opts->host == NULL) {
+                fputs("Error processing host\n", stderr);
```
[the remark this answers is not in the document]
Comment (adonSh) [reply src/args.c]: Fixed in 9fa876d.

File: readme.md [hunk 7]
```diff
@@ -92,7 +101,7 @@ effectively.
 ## help text
 
 ```
-usage: ashuffle -h -n { ..opts.. } [-e PATTERN ...] [-o NUMBER] [-f FILENAME]
+usage: ashuffle -? -n { ..opts.. } [-e PATTERN ...] [-o NUMBER] [-f FILENAME]
```
[the remark this answers is not in the document]
Comment (adonSh) [reply readme.md]: Fixed in 9fa876d.

**Comment by joshkunz** [comment]
Thanks for the PR! Merged as v1.2.0 with a couple extra small fixes that didn't seem to warrant a review.

**Pull request merged**